components/encoder/incremental/incremental_encoder.go

```diff
@@ -227,9 +231,10 @@ func (e *Encoder) Start(ctx context.Context) {
 	e.activeBackgroundWorkers.Add(1)
 
 	utils.ManagedGo(func() {
-		defer e.A.RemoveCallback(chanA)
-		defer e.B.RemoveCallback(chanB)
 		for {
+			// Remove the callbacks added by the interrupt stream.
+			defer board.RemoveCallbacks(e.board, e.interrupts, ch)
```

Comment:
This is going to defer the removal every time we go through the for loop. Put this up one line, where the old defer statement was.

Reply:
my bad thanks for catching!